apps/openchallenges/edam-etl/src/main.py

```diff
@@ -1,16 +1,68 @@
-"""Extract, transform and load EDAM concepts"""
-
+import pandas as pd
+import requests
 from os import getenv
 
 # Get config from the environment variables
 oc_db_url = getenv("OC_DB_URL")
+version = getenv("OC_DB_VERSION")
+print(f"EDAM Version: {version}")
 print(f"OC DB URL: {oc_db_url}")
 
-# TODO Download EDAM concepts from GitHub or S3 bucket (CSV file)
-print("[TODO] Download EDAM concepts from GitHub or S3 bucket (CSV file)")
 
-# TODO Process the EDAM concepts
-print("[TODO] Process the EDAM concepts")
+# Download EDAM concepts from GitHub or S3 bucket (CSV file)
+def download_edam_csv(url, version):
+    print("Downloading the EDAM concepts from GitHub (CSV file)...")
+    try:
+        response = requests.get(url)
+        response.raise_for_status()  # Raise an exception for bad response
+        with open(f"EDAM_{version}.csv", "wb") as f:
+            f.write(response.content)
+        print("EDAM concepts downloaded successfully.")
+    except requests.RequestException as e:
+        print(f"Error downloading EDAM concepts: {e}")
+
+
+# Transform the CSV to a DataFrame with indices starting from 0
+def transform_to_dataframe(version):
+    print("Processing the EDAM concepts...")
+    try:
+        df = pd.read_csv(f"EDAM_{version}.csv")
+        df["id"] = df.index
+        # Select and rename specific columns
+        df = df[["id", "Class ID", "Preferred Label"]]
+        df = df.rename(
+            columns={"Class ID": "class_id", "Preferred Label": "preferred_label"}
+        )
+        print("EDAM concepts processed successfully.")
+        return df
+    except FileNotFoundError:
+        print(f"File EDAM_{version}.csv not found.")
+    except Exception as e:
+        print(f"Error processing EDAM concepts: {e}")
+
+
+# Gather data about the EDAM ontology
+def print_info_statistics(df):
+    if df is not None:
+        print(f"Number of Concepts Transformed: {len(df)}")
+        print(f"Column names: {df.columns.tolist()}")
+        print("Statistics:")
+        # Set the display options to show only 2 decimal places
+        pd.set_option("display.float_format", "{:.0f}".format)
+        print(df.describe())
+    else:
+        print("No data available.")
+
+
+def main():
+    url = "https://github.com/edamontology/edamontology/raw/main/releases/EDAM_1.25.csv"
```

Comment:
Perhaps interpolate `VERSION` into this url as well?

```suggestion
    url = f"https://github.com/edamontology/edamontology/raw/main/releases/EDAM_{VERSION}.csv"
```

That way, we won't need to manually update this when the time comes!

Reply:
interpolated version into the url